heareval/predictions/task_predictions.py

```diff
@@ -253,6 +254,40 @@ def validation_step(self, batch, batch_idx):
     def test_step(self, batch, batch_idx):
         return self._step(batch, batch_idx)
 
+    def log_scores(self, name: str, score_args):
+        """Logs the metric score value for each score defined for the model"""
+        assert hasattr(self, "scores"), "Scores for the model should be defined"
+        end_scores = {}
+        for score in self.scores:
+            score_ret = score(*score_args)
+            validate_score_return_type(score_ret)
+            # If the returned score is a tuple, store each subscore as separate entry
+            if isinstance(score_ret, tuple):
+                # The first score in the returned tuple will be used
+                # as primary score for this metric
```

Comment:
Wait. No. WTF.

We don't have a primary score per metric. We have a primary score AKA optimization criterion AKA target score for each *task*.

Reply:
by priamry score at the metric level, i meant the fms will be primary in any of the optimisation or will behave as the target

Reply:
added information for this in `validate_score_return_type`